bit_manipulation/count_distinct_primes_from_binary_string.cpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file count_distinct_primes_from_binary_string.cpp
+ * @brief Count distinct primes formed from binary strings using allowed operations.
+ * 
+ * @author Rudraksh Tank
+ * @date July 2025
+ *
+ * @details
+ * Given a binary string, the task is to count how many distinct prime decimal numbers
+ * can be formed by:
+ * - Swapping any two characters (makes position irrelevant)
+ * - Changing any '1' to '0' (not the reverse)
+ *
+ * Efficient solution using bit manipulation and Sieve of Eratosthenes.
+ *
+ * Tags: Bit Manipulation, Prime Numbers, Combinatorics, Greedy, Bitmask
+ */
+
+#include <iostream>
+#include <vector>
+#include <unordered_set>
+#include <algorithm>
+
+const int MAX = 1e6;
+std::vector<bool> is_prime;

```

Comment:
```suggestion
static std::vector<bool> is_prime;

```

Reply:
Since you want to pre compute primes why not make this a constexpr? and assign in it directly? instead of a call from main. Also afaik vectors cant be used as constexpr in c++17 hence it might be preferred if you use std::array instead.

if any of this sounds confusing let me know I can explain or direct you to resources to learn about this feature of c++